### PR TITLE
TE-10560 Documentation page for Enable queue task flow in New Relic

### PR DESCRIPTION
**Pull request by yaroslav-spryker**
## PR Description

Documentation page for Enable queue task flow in New Relic

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/enable-queue-task-flow-in-new-relic.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+---
+
+# Problem statement
+By default `spryker/monitoring` module groups all console command New Relic transactions by command’s name e.g. `ooms:check-timeout`.
+
+For most cases this transaction naming strategy would work, but there are cases when this behaviour needs to be overridden. 
+
+Let’s consider command `queue:task:start [queue_name]`. With default naming strategy all calls of the command with all different queues will be named the same `queue:task:start` thus making transaction data aggregated under the same name for all types for queue processors.
+
+In New Relic panel this will leads to the situation as on the following illustration:
```
Comment (yaroslav-spryker) [inline docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/enable-queue-task-flow-in-new-relic.md:15]: Please add an image

image

File: docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/enable-queue-task-flow-in-new-relic.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+> :info: Transaction data is harvested and can be queried by attribute with NRQL.
+
+# Goal
+
+Enable `spryker/monitoring` module to aggregate `queue:task:start` transactions not only by command name, but command and first argument. Result of the desired behaviour is illustrated on the following screenshot.
+
```
Comment (yaroslav-spryker) [inline docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/enable-queue-task-flow-in-new-relic.md:22]: Please add an image

image

File: docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/enable-queue-task-flow-in-new-relic.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+1. Provide an advanced extendable infrastructure for [New relic monitoring transaction](https://docs.newrelic.com/docs/apm/transactions/intro-transactions/transactions-new-relic-apm/) naming strategies that will allow to implement, configure and override default transaction naming behaviour. 
+2. Implement strategy that will allow for configured console commands apply naming that groups transaction by command name and first argument.
+3. Enable new strategy for queue:task:start at the project level.
+
+# Project enablement
+For enabling queue task flow in New Relic, please see “Spryker elastic computing” document.
+
+# Implementation details
+Class diagram
```
Comment (yaroslav-spryker) [inline docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/enable-queue-task-flow-in-new-relic.md:33]: Please add an image

image

File: docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/enable-queue-task-flow-in-new-relic.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+By default, the `spryker/monitoring` module groups all New Relic transactions generated from a console command by the command’s name. For example, `ooms:check-timeout`.
+
+This strategy works in most cases, but you might want to override this behavior. For example, `queue:task:start {queue_name}`. With the default naming strategy, all the calls of the command with different queues are named `queue:task:start`. Transaction data is aggregated under the same name for all types of queue processors.
+
+In New Relic dashboard, this looks as follows:
+
+![New relic one]()
+
+> :info: Transaction data is harvested and can be queried by attribute with NRQL. === What are we trying to explain by this note?
```
Comment (andriitserkovnyi) [inline docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/enable-queue-task-flow-in-new-relic.md:16]: What are we trying to explain by this note?

File: docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/enable-queue-task-flow-in-new-relic.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
+---
+title: Enable queue task flow in New Relic
```
Comment (andriitserkovnyi) [inline docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/enable-queue-task-flow-in-new-relic.md:2]: What do you mean by `queue task flow`? The doc seems to explain something different

Comment (qlmmlp) [reply docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/enable-queue-task-flow-in-new-relic.md:2]: In general this references to the queue processing in P&S infrastructure.

File: docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/enable-queue-task-flow-in-new-relic.md (hunk 4 above)
Comment (qlmmlp) [reply docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/enable-queue-task-flow-in-new-relic.md:16]: Although transactions are collected under the same name, the data of particular queue processing transaction is still accessible but through querying it with NRQL.  

=== What are we trying to explain by this note?
this needs to be removed.

**Review by andriitserkovnyi — APPROVED** [review]

**Pull request merged**